src/sinks/datadog/metrics/encoder.rs

```diff
@@ -772,9 +781,16 @@ mod tests {
         let result = encoder.finish();
         assert!(result.is_ok());
 
-        let (_payload, mut processed) = result.unwrap();
+        let (payload, mut processed, raw_bytes) = result.unwrap();
         assert_eq!(processed.len(), 1);
         assert_eq!(expected, processed.pop().unwrap());
+        assert_eq!(100, payload.len());
+
+        // The payload is:
+        // {"metric":"basic_counter","type":"count","interval":null,"points":[[1651664333,3.14]],"tags":[]}
+        // which comes to a total of 98 bytes.
+        // There are extra bytes that make up the header and footer. These should not be included in the raw bytes.
+        assert_eq!(96, raw_bytes);
```

Comment:
The header and footer bytes do need to be included.  They represent data that is part of the compressed request.